spec/controllers/queue_items_controller_spec.rb

```diff
@@ -107,4 +116,178 @@
       expect(response).to redirect_to sign_in_path
     end
   end
+
+  describe 'PATCH update_queues' do
+    context 'for updating positions' do
+      context 'for authenticated users' do
+        it 'redirects to my queue path' do
+          user = Fabricate(:user)
+          session[:user_id] = user.id
+        end
+
+        it 'changes the position of two queue items' do
+          user = Fabricate(:user)
+          session[:user_id] = user.id
+          queue_item1 = Fabricate(:queue_item, id: 1, position: 1)
+          queue_item2 = Fabricate(:queue_item, id: 2, position: 2)
+          original_queue_item2_position = queue_item2.position
+          patch :update_queues, queue_ids_with_positions: { queue_item1.id.to_s => 2,  queue_item2.id.to_s => 1 }
+          expect(QueueItem.first.position).to eq(2)
+        end
+
+        it 'does not change the position of several list items if two positions are same' do
+          user = Fabricate(:user)
+          session[:user_id] = user.id
+          queue_item1 = Fabricate(:queue_item, id: 1, position: 1)
+          queue_item2 = Fabricate(:queue_item, id: 2, position: 2)
+          original_queue_item2_position = queue_item2.position
+          patch :update_queues, queue_ids_with_positions: { queue_item1.id.to_s => 1,  queue_item2.id.to_s => 1 }
+          expect(queue_item2.position).to eq(2)
```

Comment:
You can use `change` RSpec matcher.

`expect { patch :update_queues ....... }.to_not change { queue_item2.position }`